### DISCUSSION TITLE
Cannot use child class from VRoidPlayerControllerBase

### DISCUSSION THREAD

**Discussion by ToffelsKater** [pixiv/vroid-sdk-developers#160] [Q&A] [answered]
Hello when trying to use Multiplay Function and making a child class from VRoidPlayerControllerBase, multiplay does not work and vroid characters are not getting synced, only when the original VRoidPlayerControllerBase controller is used it will work

Any ideas?

**Comment by mameoPix** [comment]
It appears that the DownloadLicenseIdArray in the PlayerController to which it is inherited is empty.
I was able to confirm that it works by updating it using OnSwapPlayerControllers in GameMode.
![image](https://github.com/pixiv/vroid-sdk-developers/assets/104663155/6f571a42-35f7-4865-be9d-aecf1d83c9f6)

**Comment by mameoPix** [comment, marked as answer]
A simpler solution would be to cast the PlayerController in SeamlessTravelFrom to get the DownloadLicenseIdArray, as shown below.
```
void AVRoidPlayerControllerBase::SeamlessTravelFrom(APlayerController* OldPC)
{
	Super::SeamlessTravelFrom(OldPC);

	if (const auto OldVRoidPC = Cast<AVRoidPlayerControllerBase>(OldPC))
	{
		DownloadLicenseIdArray = OldVRoidPC->DownloadLicenseIdArray;
	}
}
```

**Reply by ToffelsKater** [reply]
Thanks I'll try it out, in which class is the function you showed in?

**Reply by mameoPix** [reply]
SeamlessTravelFrom() is overridden in the AVRoidPlayerControllerBase.
It will work by copying the above content into VRoidControllerBase.cpp :)
